### PR TITLE
GIX-2121: Fix snsOnlyProjectStore

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

snsOnlyProjectStore should not return universes of ICRC Tokens. It's used in specifically SNS only places.

# Changes

* Fix `snsOnlyProjectStore` to check whether the current universe is an ICRC Token universe.

I know that SNS universes are actually ICRC Tokens. But in our current code, that is not the case, and until we refactor to use the same, we consider them differently.

# Tests

* Add a test case for `snsOnlyProjectStore`.

Test was failing without the fix.

# Todos

- [ ] Add entry to changelog (if necessary).

Not necessary yet.

### CODE REVIEW

**Comment by lmuntaner** [comment]
Current dependencies on/for this PR:
* main
  * **PR #3890** <a href="https://app.graphite.dev/github/pr/dfinity/nns-dapp/3890?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dfinity/nns-dapp/3890?utm_source=stack-comment).

**Comment by lmuntaner** [comment]
@dskloetd @peterpeterparker please review

File: frontend/src/lib/derived/sns/sns-selected-project.derived.ts [hunk 1]
```diff
@@ -12,13 +15,16 @@ import { derived, type Readable } from "svelte/store";
...
-  isUniverseNns($selectedUniverseIdStore) ||
-  isUniverseCkBTC($selectedUniverseIdStore)
-    ? undefined
-    : $selectedUniverseIdStore
+>(
+  [selectedUniverseIdStore, isIcrcTokenUniverseStore],
+  ([$selectedUniverseIdStore, isIcrcTokenUniverse]: [Principal, boolean]) =>
+    isUniverseNns($selectedUniverseIdStore) ||
+    isUniverseCkBTC($selectedUniverseIdStore) ||
+    isIcrcTokenUniverse
```
Comment (dskloetd) [inline frontend/src/lib/derived/sns/sns-selected-project.derived.ts:25]: If I'm on a ckETH page and then I reload the page, my page universe will still be the same, but until the token universes are loaded, this logic will not know that it's not an SNS universe, right?

Can we use the aggregator data instead to know if a universe is an SNS universe?

Comment (lmuntaner) [reply frontend/src/lib/derived/sns/sns-selected-project.derived.ts:25]: Yes, that makes more sense actually.

Comment (lmuntaner) [reply frontend/src/lib/derived/sns/sns-selected-project.derived.ts]: DOne

**Comment by lmuntaner** [comment]
@dskloetd ready for another review

**Pull request merged**